src/gba/gba.c

```diff
@@ -595,6 +595,62 @@ void GBADebug(struct GBA* gba, uint16_t flags) {
 	gba->debugFlags = GBADebugFlagsClearSend(gba->debugFlags);
 }
 
+#ifdef USE_ELF
+bool GBAVerifyELFEntry(struct ELF* elf, uint32_t target) {
+	if (ELFEntry(elf) == target) {
+		return true;
+	}
+
+	struct ELFProgramHeaders ph;
+	ELFProgramHeadersInit(&ph, 0);
+	ELFGetProgramHeaders(elf, &ph);
+	size_t i;
+	for (i = 0; i < ELFProgramHeadersSize(&ph); ++i) {
+		Elf32_Phdr* phdr = ELFProgramHeadersGetPointer(&ph, i);
+		if (!phdr->p_filesz) {
+			continue;
+		}
+
+		size_t phdr_s = phdr->p_paddr;
+		size_t phdr_e = phdr_s + phdr->p_filesz;
+
+		// Does the segment contain our target address?
+		if ((target < phdr_s) || ((target + 4) > phdr_e)) {
+			continue;
+		}
+
+		// File offset to what should be the rom entry instruction
+		size_t off = phdr->p_offset + target - phdr_s;
+
+		size_t esize;
+		const char* bytes = ELFBytes(elf, &esize);
+
+		if (off >= esize) {
+			continue;
+		}
+
+		uint32_t opcode;
+		LOAD_32(opcode, 0, &bytes[off]);
```

Comment:
```suggestion
		LOAD_32(opcode, off, bytes);
```
That's what that middle field is for :P

That said, an additional alignment check is a good idea, as unaligned 4-byte loads are unspecified on ARM7TDMI (and the way it actually works is weird and not worth emulating here--we can just reject an unaligned address out of hand since that would break in actual use(